src/dev/build/tasks/generate_packages_optimized_assets.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { pipeline } from 'stream';
+import { promisify } from 'util';
+
+import gulpBrotli from 'gulp-brotli';
+// @ts-expect-error
+import gulpGzip from 'gulp-gzip';
+// @ts-expect-error
+import gulpPostCSS from 'gulp-postcss';
+// @ts-expect-error
+import gulpTerser from 'gulp-terser';
+import terser from 'terser';
+import vfs from 'vinyl-fs';
+
+import { ToolingLog } from '@kbn/dev-utils';
+import { Task, Build } from '../lib';
+
+const asyncPipeline = promisify(pipeline);
+
+const minifyKbnUiSharedDepsCSS = async (log: ToolingLog, build: Build) => {
+  const buildRoot = build.resolvePath();
```

Comment:
I think we should move the ci-stats metrics publishing to this step rather than generating them in the package, now that the actual assets aren't produced by the package anymore.

Reply:
@spalger we just need to recreate that metrics.json file content right https://github.com/elastic/kibana/pull/105971/files#diff-34ca7c938c7fb012bad5163be6b4952f48a3a1930e91200cc8710113bf7200ddL173 ?

Reply:
Alright, I think that is now done